File: 2-Ad-Hoc/3024-Mountain-Ranges/3024.cpp
#include <iostream>

int main() {
    int points, quota;
    std::cin>>points>>quota;
    int ant, curr, cnt[points] = {0}, index = 0, greatest = 0;
    bool onsequence = false;
    std::cin>>ant;
    for (int i = 0; i < points-1; ++i) {
        std::cin>>curr;
        if (curr - ant <= quota) {
            onsequence = true;
        }
        else {
            onsequence = false;
            ++index;
        }
        if (onsequence) ++cnt[index];
        ant = curr;
    }
    for (int i = 0; i < points; ++i) {
        if (greatest < cnt[i]) greatest = cnt[i];
    }
    std::cout<<greatest+1<<"\n";
    
    return 0;
}

File: 2-Ad-Hoc/3024-Mountain-Ranges/3024.py
points, quota = input().split()
points, quota = int(points), int(quota)
dots = list(map(int, input().split()))
cnt, index = [0]*points, 0
ant = dots[0]
onsequence = False
for i in range (1, points):
    curr = dots[i]
    if curr - ant <= quota:
        onsequence = True
    else:
        onsequence = False
        index += 1
    if (onsequence):
        cnt[index] += 1
    ant = curr
print (max(cnt)+1)